erc721/ERC721.vy

```diff
@@ -249,64 +245,76 @@ def safeTransferFrom(
+#      Throws if `_approved` is the current owner. (NOTE: This is not written the EIP)
 # @param _approved Address to be approved for the given NFT ID.
 # @param _tokenId ID of the token to be approved.
 @public
 def approve(_approved: address, _tokenId: uint256):
-    # get owner
-    owner: address = self._ownerOf(_tokenId)
+    owner: address = self.idToOwner[_tokenId]
+    # Throws if `_tokenId` is not a valid NFT
+    assert owner != ZERO_ADDRESS
+    # Throws if `_approved` is the current owner
     assert _approved != owner
-    # check requirements
+    # Check requirements
     senderIsOwner: bool = self.idToOwner[_tokenId] == msg.sender
-    assert (senderIsOwner or self._isApprovedForAll(owner, msg.sender))
-    # set the approval
+    senderIsApprovedForAll: bool = (self.ownerToOperators[owner])[msg.sender]
+    assert (senderIsOwner or senderIsApprovedForAll)
+    # Set the approval
     self.idToApprovals[_tokenId] = _approved
     log.Approval(owner, _approved, _tokenId)
 
 
 # @dev Enables or disables approval for a third party ("operator") to manage all of
 #      `msg.sender`'s assets. It also emits the ApprovalForAll event.
-# @notice This works even if sender doesn't own any tokens at the time.
+#      NOTE: This works even if sender doesn't own any tokens at the time.
 # @param _operator Address to add to the set of authorized operators.
 # @param _isApproved True if the operators is approved, false to revoke approval.
 @public
 def setApprovalForAll(_operator: address, _approved: bool):
-    assert _operator != ZERO_ADDRESS
     self.ownerToOperators[msg.sender][_operator] = _approved
     log.ApprovalForAll(msg.sender, _operator, _approved)
 
 
 ### MINT & BURN FUNCTIONS ###
 
 # @dev Function to mint tokens
+#      Throws if `msg.sender`` is not the minter.
+#      Throws if `_to` is zero address.
 # @param to The address that will receive the minted tokens.
 # @param tokenId The token id to mint.
 # @return A boolean that indicates if the operation was successful.
 @public
 def mint(_to: address, _tokenId: uint256) -> bool:
+    # Throws if `msg.sender`` is not the minter
     assert msg.sender == self.minter
+    # Throws if `_to` is zero address
     assert _to != ZERO_ADDRESS
     self._addTokenTo(_to, _tokenId)
     log.Transfer(ZERO_ADDRESS, _to, _tokenId)
     return True
 
 
 # @dev Burns a specific ERC721 token.
+#      Throws unless `msg.sender` is the current owner, an authorized operator, or the approved
+#      address for this NFT.
+#      Throws if `_tokenId` is not a valid NFT.
 # @param tokenId uint256 id of the ERC721 token to be burned.
 @public
 def burn(_tokenId: uint256):
+    # Check requirements
     assert self._isApprovedOrOwner(msg.sender, _tokenId)
-    owner: address = self._ownerOf(_tokenId)
+    owner: address = self.idToOwner[_tokenId]
+    # Throws if `_tokenId` is not a valid NFT
+    assert owner == ZERO_ADDRESS
```

Comment:
Is this intended to check `assert owner != ZERO_ADDRESS` ?

Reply:
:dogeza: